### PR TITLE
License macro fixes

### PR DESCRIPTION
**Pull request by lawruble13**

### CODE REVIEW

**Comment by lawruble13** [comment]
To clarify something here: the macro `rocm_setup_license` is not intended to be called by users, it is intended to be used internally to `rocm_create_package`, in a similar fashion to `rocm_set_comp_cpackvar`. It is broken out into a macro only to ensure that `rocm_create_package` doesn't become too long.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## [0.7.1]
 ### Added
 - `CLANG_TIDY_USE_COLOR` variable added to control the color output from `clang-tidy`, by default this is `On`.
+- If `CPACK_RESOURCE_FILE_LICENSE` is not set, `rocm_create_package` will now automatically attempt to find a LICENSE, LICENSE.txt, or LICENSE.md file in the top-level CMake source directory. If one is found, it is set up as the license file for the package.
```
Comment (pfultz2) [inline CHANGELOG.md:6]: Can this be added to `rocm_create_package` docs as well?

**Comment by pfultz2** [comment]
Perhaps we could infer `CPACK_RPM_PACKAGE_LICENSE` from the license file. Maybe for a later PR.

**Comment by lawruble13** [comment]
> Perhaps we could infer `CPACK_RPM_PACKAGE_LICENSE` from the license file. Maybe for a later PR.

There was a discussion in #64 about attempting to do so, we didn't go ahead with it as incorrectly detecting the license type could cause legal issues.

**Comment by lawruble13** [comment]
@pfultz2 Is this documentation acceptable?

**Pull request merged**